eng/pipelines/libraries/helix.yml

```diff
@@ -28,3 +27,5 @@ steps:
     env:
       SYSTEM_ACCESSTOKEN: $(System.AccessToken) # We need to set this env var to publish helix results to Azure Dev Ops
       HelixTargetQueues: ${{ parameters.helixQueues }} # Pass queues to MSBuild as env var to avoid need of escaping them
+      ${{ if ne(parameters.framework, '') }}:
+        TargetGroup: ${{ parameters.framework }} # Pass as env var as we can't condition otherwise, we can't pass empty to set default value.
```

Comment:
Can you please elaborate?

Reply:
Of course. 

So basically we're now not passing down netcoreapp through parameters.framework, and we're leaving parameters.framework empty. Which means that we're relying on our MSBuild props, to set it to `NetCoreAppCurrent` when `TargetGroup == ''`. In this case, since we were not conditioning this parameter, when `parameters.framework` is empty, we would be passing `/p:TargetGroup=` via command line to `sendtohelix.proj`. Since it is a global property via cmd, when we default to `NetCoreAppCurrent` it won't be set because you can't override a global property in MSBuild. So we need to condition this when we flow it to only be flown when framework != ''. 

However, since this is a step template, we can't define variables, nor condition the argument in the -script task invocation itself because yml doesn't support that. So the only way to pass it down with a condition is via env var.

Reply:
As discussed in today's sync it would be great and necessary to not use TargetGroup to detect if we are running an AllConfigurations build. This probably impact your changes here.

Reply:
It is still needed for netfx though, as we need to zip up the testhost which is dependent on the BuildConfiguration. 

I will update to not depend on TargetGroup=allConfigurations, but we still need the BuildConfiguration for netfx...

Reply:
Sounds good.

Reply:
Done.